lib/collection.js

```diff
@@ -686,7 +686,10 @@ Collection.prototype.updateOne = function(filter, update, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
 
-  const err = checkForAtomicOperators(update);
+  const err = Array.isArray(update)
+    ? update.forEach(checkForAtomicOperators)
```

Comment:
from what I can tell, this is not going to return an `err`, as `Array.prototype.forEach` returns `undefined`.

Is the goal here to get the first `checkForAtomicOperators` error on the pipeline?

Reply:
Yep, I believe that we want the first `checkForAtomicOperators` error in the pipeline.

Reply:
In that case you could build this up imperatively:
```js
let err;
if (Array.isArray(update)) {
  for (let i = 0; !err && i < update.length; i++) {
    err = checkForAtomicOperators(update);
  }
} else {
  err = checkForAtomicOperators(update);
}
```